src/config.c

```diff
@@ -10,95 +10,100 @@
 
 
 s_fidoconfig *config;
-char *logFileName=NULL;
-char *fidoConfigFile=NULL;
-int fidocfg_in_env=0;
+char *fidoConfigFile = NULL;
+int fidocfg_in_arg = 0;
 char *versionStr;
 
 void Usage()
 {
     printf("\nUsage: bsopack [-c fidconfig] [options]\n");
     printf("Options:\n");
     printf("\t-h or --help  This help screen\n\n");
+
+    if (fidocfg_in_arg)
+    {
+        nfree(fidoConfigFile);
+    }
 }
 
 void getOpts(int argc, char **argv)
 {
     int i;
 
-    fidoConfigFile=getenv("FIDOCONFIG");
-    if (fidoConfigFile!=NULL)
-        fidocfg_in_env=1;
+    fidoConfigFile = getenv("FIDOCONFIG");
 
-    for (i=1;i<argc;i++)
+    for (i = 1; i < argc; i++)
     {
-        if (argv[i][0]=='-')
+        if (!strcmp(argv[i], "-h") || !strcmp(argv[i], "--help"))
+        {
+            Usage();
+            exit(0);
+        }
+
+        if (!strcmp(argv[i], "-c"))
         {
-            if ((strchr(argv[i], 'h'))||(!strcmp(argv[i]+1, "-help")))
+            if (fidocfg_in_arg)
             {
+                fprintf(stderr, "Duplicate command line parameter -c.\n");
                 Usage();
-                exit(0);
+                exit(-1);
             }
-            if (strchr(argv[i], 'c'))
+
+            if (++i >= argc)
             {
-                if (i+1>=argc)
-                {
-                    fprintf(stderr, "Incorrect command line parameter.\n");
-                    Usage();
-                    exit(-1);
-                }
-                fidoConfigFile=(char *)smalloc(strlen(argv[i+1]+1));
-                sprintf(fidoConfigFile, "%s", argv[i+1]);
-                fidocfg_in_env=0;
-                ++i;
+                fprintf(stderr, "Incorrect command line parameter.\n");
+                Usage();
+                exit(-1);
             }
+
+            fidoConfigFile = (char *)smalloc(strlen(argv[i]));
+            sprintf(fidoConfigFile, "%s", argv[i]);
```

Comment:
Here, only `strlen(argv[i])` is printed and `fidoConfigFile[strlen(argv[i]) + 1]` is left undefined. We can use `snprintf`:
`snprintf(fidoConfigFile, strlen(argv[i]) + 1, "%s", argv[i]);`

Reply:
Hmm, it was old code, not mine... Ok, I'll fix it, thanks.